core/build.gradle.kts

```diff
@@ -73,246 +88,204 @@ kotlin {
                 }
             }
             common("androidNative") {
+                // Tier 3
                 target("androidNativeArm32")
                 target("androidNativeArm64")
                 target("androidNativeX86")
                 target("androidNativeX64")
             }
         }
-        // Tier 3
         common("windows") {
+            // Tier 3
             target("mingwX64")
         }
     }
 
-    jvm {
-        attributes {
-            attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
+    targets.withType<KotlinNativeTarget> {
+        compilations["test"].compileTaskProvider.configure {
+            compilerOptions.freeCompilerArgs.add("-trw")
         }
-        compilations.all {
-            // Set compilation options for JVM target here
+        if (konanTarget.family == Family.MINGW) {
+            compilations["test"].cinterops {
+                create("modern_api") {
+                    defFile("$projectDir/windows/test_cinterop/modern_api.def")
+                    headers("$projectDir/windows/test_cinterop/modern_api.h")
+                }
+            }
         }
-
     }
 
-    js {
+    @OptIn(ExperimentalWasmDsl::class)
+    wasmJs {
```

Comment:
Why wasmJs was moved higher? It's the most recent target amongst all, so we put it last by convention